### PR TITLE
Adapt interpolation object for GPU runs. 

### PR DESCRIPTION
**Pull request by akshaysridhar**

### CODE REVIEW

File: src/solver/type_getters.jl [hunk 1]
```diff
@@ -135,17 +136,21 @@ function get_spaces(parsed_args, params, comms_ctx)
         warp_function = nothing
     elseif topography == "Earth"
         data_path = joinpath(topo_elev_dataset_path(), "ETOPO1_coarse.nc")
+        array_type = ClimaComms.array_type(ClimaComms.device())
```
Comment (charleskawczynski) [inline src/solver/type_getters.jl:139]: Why are we getting the device this way? Shouldn't we already have it somewhere?

Comment (akshaysridhar) [reply src/solver/type_getters.jl:139]: Something like `ctx = ClimaComms.context(axes(Y.c))` to then get `ctx.device` perhaps based on this ```ClimaComms.context(device=device())``` in ClimaComms perhaps ? ; I'll check if we have this stored somewhere already in`model` or `type` getters.

Comment (akshaysridhar) [reply src/solver/type_getters.jl]: @charleskawczynski Fixed (device is accessible in ctx) and is available at this level without calling the ClimaComms `device` function

**Comment by akshaysridhar** [comment]
@charleskawczynski If the compat entry update + re-built Manifests look reasonable I can merge this.

File: Project.toml [hunk 2]
```diff
@@ -42,6 +43,7 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
 YAML = "ddb6d928-2868-570f-bddf-ab3f9cf99eb6"
 
 [compat]
+Adapt = "3.7"
```
Comment (Sbozzolo) [inline Project.toml:46]: Why this version?

Would this work?
```suggestion
Adapt = "3, 4"
```

Comment (charleskawczynski) [reply Project.toml:46]: This won't do anything yet because RRTMGP still restricts us

**Review by charleskawczynski — APPROVED** [review]

File: Project.toml (hunk 2 above)
Comment (akshaysridhar) [reply Project.toml:46]: Yeah my intention was to pin it to a known working version, here, `3.7` (given the issues we seemed to be having elsewhere with CUDA / Adapt upgrades)

File: perf/Project.toml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 [deps]
+Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
```
[the remark this answers is not in the document]
Comment (Sbozzolo) [reply perf/Project.toml:2]: But this is the `perf` env. `Adapt` is never used directly here.

[the remark this answers is not in the document]
Comment (Sbozzolo) [reply perf/Project.toml:2]: Ops, got it.

**Pull request merged**